schutzbot/ci_details.sh

```diff
@@ -42,19 +42,22 @@ echo "List of installed packages:"
 rpm -qa | sort
 echo "------------------------------------------------------------------------------"
 
-if rpm --quiet -q python36; then
-    echo -e "\n FAIL: python36 is installed, see #794 ..."
-    exit 1
-else
-    echo -e "\n PASS: python36 not insalled"
+# gcp runners don't use cloud-init and some of the images have python36 installed
+if [[ "$RUNNER" != *"gcp"* ]];then 
+    if rpm --quiet -q python36; then
+        echo -e "\n FAIL: python36 is installed, see #794 ..."
+        exit 1
+    else
+        echo -e "\n PASS: python36 not insalled"
+    fi
+
+    # Ensure cloud-init has completely finished on the instance. This ensures that
+    # the instance is fully ready to go.
+    while true; do
+      if [[ -f /var/lib/cloud/instance/boot-finished ]]; then
+        break
+      fi
+      echo -e "\n🤔 Waiting for cloud-init to finish running..."
+      sleep 5
+    done
 fi
-
-# Ensure cloud-init has completely finished on the instance. This ensures that
-# the instance is fully ready to go.
-while true; do
-  if [[ -f /var/lib/cloud/instance/boot-finished ]]; then
-    break
-  fi
-  echo -e "\n🤔 Waiting for cloud-init to finish running..."
-  sleep 5
-done
```

Comment:
The commit log for dd0fc32b02c47402f284646d1a2fab355635eead says

>    We need to wait for cloud-init to be completely finished since it is
>    registering RHEL instances with RHN and preparing a swap file.

I'm not sure if this is an issue with GCP but worth investigating a bit more. 

- How do we register GCP systems to RHN if there is no cloud-init? 
- How do we know the VM has booted up completely and ready for testing ?

Reply:
These images have some google guest packages which are described a bit here https://github.com/osbuild/osbuild-composer/blob/main/image-types/rhel8/google-gce.md#implementation-choices-osbuild-composer-gce-rhel-guest-image 

We register them with subscription-manager during the `prepare` step https://github.com/osbuild/gitlab-ci-terraform-executor/blob/main/prepare#L75 and it waits for the registration to finish I believe.

And how do we know VM is ready I'm not sure. Maybe terraform has a way to tell? It does wait until the provisioning is finished and then it waits for the ssh connection to be available, that sounds like enough to me... But I can try taking a look at how the google guest stuff works and if there is something we can check similar to `cloud-init`

Reply:
Since this comes from the age of Jenkins I feel like it might have been some issue with the way runners were provisioned then...